src/redux/categories/categoriesSlice.js

```diff
@@ -0,0 +1,55 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  loading: false,
+  categories: [],
+  status: 'Under construction', // Check if the '' needs to be in the array or just returned
+  error: '',
+};
+
+const categoriesSlice = createSlice({
+  name: 'category',
+  initialState,
+  reducers: {
+    addCategory: (state, action) => {
+      state.categories.push(action.payload);
+    },
+    removeCategory: (state, action) => {
+      const idToRemove = action.payload;
+      state.categories = state.categories.filter((category) => category.id !== idToRemove);
+    },
+    updateCategory: (state, action) => {
+      const {
+        id, title, author, category,
+      } = action.payload;
+      const categoryIndex = state.categories.findIndex(
+        (category) => category.id === categoryIndex,
+      );
+      if (categoryIndex !== -1) {
+        state.categories[categoryIndex] = {
+          id, title, author, category,
+        };
+      }
+    },
+    checkCategory: (state, action) => {
+      action.payload = 'Under construction';
+      return action.payload;
+      /*
+      state.categories = 'Under construction';
+      return state.categories */
+    },
+  },
```

Comment:
- [ ] As part of the project requirements you only need 1 reducer which you already have the `checkCategory` and make sure that the reducer checks the status from the action payload before returning the message as shown below 👇.
```jsx
reducers: {
    checkCategory: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },

```